### ISSUE TITLE
Issue when detecting root page of non-existing language

### ISSUE THREAD

**Issue by aschempp** [contao/core#8142]
I have found a funny issue thanks to my error reporting tool.

```
More than one page matches page ID "index". (http://example/en/)
```

For the given domain, there is a root page for language `de` and `fr`. German is fallback, and language in URL is enabled.

Somehow a visitor added `/en/` to the URL. The system seems unable to find the fallback page, it should redirect to german in my opinion.

PS: this was Contao 4 but I'm pretty sure it's the same problem in Contao 3.

**Comment by leofeyer** [comment]
The issue seems to be specific to Contao 4. The error message does not exist in Contao 3.

**Comment by leofeyer** [comment]
Fixed in 5a523ce7b2bdf814936c321f74ed317b2ed1fb03.
